### DISCUSSION TITLE
Calling an action directly from a server component

### DISCUSSION THREAD

**Discussion by josiahwiebe** [next-safe-action/next-safe-action#298] [Q&A]
All of the examples illustrate how you can call actions from a **client component**, but when you try to call an action in a **server component**, you get a type error.

This won't work in a server component:
```jsx
<form action={logout}>
  <button type='submit'>Log out</button>
</form>
```

but this will work:
```jsx
<form action={async () => {
    'use server'
    await logoutAction()
  }}>
  <button type='submit'>Log out</button>
</form>
```

Feels like a bit of an odd workaround to me.

**Comment by benjick** [comment]
It used to work in Next.js 14 for me, but now I'm seeing type errors in 15.1. Is there no better way?
